afs/afs-storage-api/src/main/java/com/powsybl/afs/storage/DefaultListenableAppStorage.java

```diff
@@ -82,11 +84,18 @@ public String deleteNode(String nodeId) {
         return parentNodeId;
     }
 
+    /**
+     * Return an {@link OutputStream} to underlying storage which will raise a data updated event on close.
+     */
     @Override
     public OutputStream writeBinaryData(String nodeId, String name) {
-        OutputStream os = super.writeBinaryData(nodeId, name);
-        addEvent(new NodeDataUpdated(nodeId, name));
-        return os;
+        return new ProxyOutputStream(super.writeBinaryData(nodeId, name)) {
```

Comment:
Consider using ForwardingOutputStream instead of ProxyOutputStream?

Reply:
Or maybe replace our own class ForwardingOutputStream by equivalent ProxyOutputStream everywhere?